test/unit_tests/test_metadata_table.py

```diff
@@ -138,5 +138,148 @@ def test_missing_intent(self):
+       self.assertTrue(emsg in str(context.exception))
+
+   def test_bad_var_property_name(self):
+       """Test that a ddt_type = None returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_bad_var_property_name.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #print("The exception is", context.exception)
+       emsg = "Invalid variable property name, 'none', at "
+       self.assertTrue(emsg in str(context.exception))
+
+   def test_no_input(self):
+       """Test that no input returns expected error"""
+       with self.assertRaises(Exception) as context:
+           MetadataTable()
+
+       #print("The exception is", context.exception)
+       emsg = "MetadataTable requires a title"
+       self.assertTrue(emsg in str(context.exception))
+
+   def test_no_header_type(self):
+       """Test that __init__ with type_in=None returns expected error"""
+       with self.assertRaises(Exception) as context:
+           MetadataTable(parse_object=None, title="something", type_in=None, \
+               module=None, var_dict=None, known_ddts=None, logger=None)
+
+       #print("The exception is", context.exception)
+       emsg = "MetadataTable requires a header type"
+       self.assertTrue(emsg in str(context.exception))
+
+   def test_bad_header_type(self):
+       """Test that __init__ with type_in=banana returns expected error"""
+       with self.assertRaises(Exception) as context:
+           MetadataTable(parse_object=None, title="something", type_in="banana", \
+               module=None, var_dict=None, known_ddts=None, logger=None)
+
+       #print("The exception is", context.exception)
+       emsg = "Invalid metadata table type, 'banana'"
+       self.assertTrue(emsg in str(context.exception))
+
+   def test_no_module(self):
+       """Test that __init__ with module=None returns expected error"""
+       with self.assertRaises(Exception) as context:
+           MetadataTable(parse_object=None, title="something", type_in="scheme", \
+               module=None, var_dict=None, known_ddts=None, logger=None)
+
+       #print("The exception is", context.exception)
+       emsg = "MetadataTable requires a module name"
+       self.assertTrue(emsg in str(context.exception))
+
+   def test_bad_1st_ccpp_arg_table(self):
+       """Test that first arg table named ccpp-farg-table returns expected error"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_bad_1st_arg_table_header.meta")
+
+       with self.assertRaises(Exception) as context:
+           MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       print("The exception is", context.exception)
+       #emsg = "Unknown DDT type, banana, at "
+       #self.assertTrue(emsg in str(context.exception))
```

Comment:
I do not think these should not be commented.
I think you should put in the error message you expect (same as in the next test).